src/argilla/client/feedback/dataset/remote/dataset.py

```diff
@@ -141,27 +250,187 @@ def __init__(
+
+        self._client = client  # Required to be able to use `allowed_for_roles` decorator
+        self._id = id
+        self._name = name
+        self._workspace = workspace
+        self._created_at = created_at
+        self._updated_at = updated_at
+
+        self._records = RemoteFeedbackRecords(dataset=self)
+
+    @property
+    def records(self) -> RemoteFeedbackRecords:
+        """Returns an instance of `RemoteFeedbackRecords` that allows you to iterate over
+        the records in the dataset. The records are fetched from Argilla on the fly and
+        not stored in memory. You can also iterate over the records directly from the
+        dataset instance.
+        """
+        return self._records
+
+    @property
+    def id(self) -> "UUID":
+        """Returns the ID of the dataset in Argilla."""
+        return self._id
+
+    @property
+    def name(self) -> str:
+        """Returns the name of the dataset in Argilla."""
+        return self._name
+
+    @property
+    def workspace(self) -> "Workspace":
+        """Returns the workspace the dataset belongs to in Argilla."""
+        return self._workspace
+
+    @property
+    def url(self) -> str:
+        """Returns the URL of the dataset in Argilla."""
+        return f"{self._client.base_url}/dataset/{self.id}/annotation-mode"
+
+    @property
+    def created_at(self) -> datetime:
+        """Returns the datetime when the dataset was created in Argilla."""
+        return self._created_at
+
+    @property
+    def updated_at(self) -> datetime:
+        """Returns the datetime when the dataset was last updated in Argilla."""
+        return self._updated_at
+
+    def __repr__(self) -> str:
+        """Returns a string representation of the dataset."""
+        return (
+            f"<FeedbackDataset id={self.id} name={self.name} workspace={self.workspace}"
+            f" url={self.url} fields={self.fields} questions={self.questions}"
+            f" guidelines={self.guidelines}>"
+        )
+
+    def __len__(self) -> int:
+        """Returns the number of records in the dataset."""
+        return self._records.__len__()
+
+    def __iter__(self) -> Iterator[RemoteFeedbackRecord]:
+        """Returns an iterator over the records in the dataset."""
+        yield from self._records
+
+    def __getitem__(self, key: Union[slice, int]) -> Union[RemoteFeedbackRecord, List[RemoteFeedbackRecord]]:
+        """Returns the record(s) at the given index(es).
+
+        Args:
+            key: The index or slice to retrieve.
+
+        Returns:
+            The record(s) at the given index(es).
+        """
+        return self._records.__getitem__(key)
+
+    def sort_by(self, sort: List[SortBy]) -> "RemoteFeedbackDataset":
+        """Sorts the current `RemoteFeedbackDataset` based on the given sort fields and orders."""
+        sorted_dataset = self.__copy_from(self)
+        sorted_dataset._records = RemoteFeedbackRecords.copy_from(sorted_dataset, sort_by=sort)
```

Comment:
I think the previous approach using the `records_cls` as a global class attribute may be more consistent here as one should just `cls.records_cls = ...` and then instantiate `cls(...)`

Reply:
We only need one single `RemoteFeedbackRecords` implementation for this kind of dataset, so the conditional record class is not needed anymore. Also, my feeling reading the code was that the logic was diluted in several places and it was hard to track what was going on.